target/linux/ramips/dts/RE650.dts

```diff
@@ -0,0 +1,174 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7621.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,re650-v1", "mediatek,mt7621-soc";
+	model = "TP-Link RE650 v1";
+
+	aliases {
+		led-boot = &power;
+		led-failsafe = &power;
+		led-running = &power;
+		led-upgrade = &power;
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x8000000>;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,57600";
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		power: power {
+			label = "re650-v1:blue:power";
+			gpios = <&gpio0 22 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan2g {
+			label = "re650-v1:blue:wlan2g";
+			gpios = <&gpio0 23 GPIO_ACTIVE_LOW>;
+		};
+
+		wlan5g {
+			label = "re650-v1:blue:wlan5g";
+			gpios = <&gpio0 24 GPIO_ACTIVE_LOW>;
+		};
+
+		wps_red {
+			label = "re650-v1:red:wps";
+			gpios = <&gpio0 25 GPIO_ACTIVE_HIGH>;
+		};
+
+		wps_blue {
+			label = "re650-v1:blue:wps";
+			gpios = <&gpio0 26 GPIO_ACTIVE_HIGH>;
+		};
+
+		eth_act {
+			label = "re650-v1:green:eth_act";
+			gpios = <&gpio0 28 GPIO_ACTIVE_LOW>;
+		};
+
+		eth_link {
+			label = "re650-v1:green:eth_link";
+			gpios = <&gpio0 29 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
```

Comment:
Question to other reviewers: Is gpio-keys available on this target?

Reply:
Since commit 110daa16e44bb53941a18396ab6a7d9fdd1fa362 the gpio-keys should actually work on mt7621.

Reply:
Okay, so then you may/should replace gpio-keys-polled by gpio-keys and remove poll-interval ... :-)

Reply:
Done. I also added debounce-interval of 60 msec. Looking good with the exception of that "power" button (gpio 32). I can't see that gpio changing level when pressed. I picked the number from the GPL source, but it might be something different. Reset, wps and led buttons are fine.

```
root@OpenWrt:~# cat /proc/interrupts |grep keys
 26:          6          0          0          0      GPIO  18  keys
 30:          4          0          0          0      GPIO  30  keys
 31:          0          0          0          0      GPIO  31  keys
 32:          0          0          0          0      GPIO  32  keys
```